src/main.cc

```diff
@@ -86,6 +87,28 @@ class Spellchecker : public Nan::ObjectWrap {
     }
   }
 
+  static NAN_METHOD(CheckSpellingCallback) {
+    Nan::HandleScope scope;
+    if (info.Length() < 2) {
+      return Nan::ThrowError("Bad argument");
+    }
+
+    Handle<String> string = Handle<String>::Cast(info[0]);
+    if (!string->IsString()) {
+      return Nan::ThrowError("Bad argument");
+    }
+
+    Nan::Callback *callback = new Nan::Callback(info[1].As<v8::Function>());
+
+    std::vector<uint16_t> *corpus = new std::vector<uint16_t>(string->Length() + 1);
```

Comment:
I think that this vector is currently leaked, because it is allocated with `new` and not deleted. You can allocate it on the stack and still avoid making a copy by having `CheckSpellingWorker` take the vector via a *move* argument:

```cpp
std::vector<uint16_t> corpus(string->Length() + 1);

// The call to `std::move` casts `corpus` to an r-value, meaning that
// it doesn't need to be used anymore, so it is safe to recycle its contents.
CheckSpellingWorker* worker = new CheckSpellingWorker(std::move(corpus), that->impl, callback);
```

```cpp
class CheckSpellingWorker : public Nan::AsyncWorker {
public:

  // The `&&` syntax means that this argument is an r-value, so its internals
  // can be recycled, rather than copying it.
  CheckSpellingWorker(std::vector<uint16_t> &&corpus, SpellcheckerImplementation* impl, Nan::Callback* callback);
}
```

```cpp
// We have to use `std::move` again here, when assigning the `corpus` field
// from the `corpus` argument.
CheckSpellingWorker::CheckSpellingWorker(
  std::vector<uint16_t> &&corpus,
  SpellcheckerImplementation* impl,
  Nan::Callback* callback
) : AsyncWorker(callback), corpus(std::move(corpus)), impl(impl) {}
```